tests/test_parser_utils.py

```diff
@@ -13,12 +14,242 @@
+        uri = Path(get_data("testdata/checker_wf/broken-wf.cwl")).resolve().as_uri()
+        cwl_obj = load_document_by_uri(uri)
+        cwl_utils.parser.utils.static_checker(cwl_obj)
+
+    with pytest.raises(ValidationException):
+        uri = Path(get_data("testdata/checker_wf/broken-wf2.cwl")).resolve().as_uri()
+        cwl_obj = load_document_by_uri(uri)
+        cwl_utils.parser.utils.static_checker(cwl_obj)
+
+    with pytest.raises(ValidationException):
+        uri = Path(get_data("testdata/checker_wf/broken-wf3.cwl")).resolve().as_uri()
+        cwl_obj = load_document_by_uri(uri)
+        cwl_utils.parser.utils.static_checker(cwl_obj)
+
+    with pytest.raises(ValidationException):
+        uri = Path(get_data("testdata/count-lines6-wf_v1_0.cwl")).resolve().as_uri()
+        cwl_obj = load_document_by_uri(uri)
+        cwl_utils.parser.utils.static_checker(cwl_obj)
+
+    with pytest.raises(ValidationException):
+        uri = Path(get_data("testdata/count-lines6-wf_v1_1.cwl")).resolve().as_uri()
+        cwl_obj = load_document_by_uri(uri)
+        cwl_utils.parser.utils.static_checker(cwl_obj)
+
+    with pytest.raises(ValidationException):
+        uri = Path(get_data("testdata/count-lines6-wf_v1_2.cwl")).resolve().as_uri()
+        cwl_obj = load_document_by_uri(uri)
+        cwl_utils.parser.utils.static_checker(cwl_obj)
+
+    with pytest.raises(ValidationException):
+        uri = (
+            Path(get_data("testdata/count-lines6-single-source-wf_v1_0.cwl"))
+            .resolve()
+            .as_uri()
+        )
+        cwl_obj = load_document_by_uri(uri)
+        cwl_utils.parser.utils.static_checker(cwl_obj)
+
+    with pytest.raises(ValidationException):
+        uri = (
+            Path(get_data("testdata/count-lines6-single-source-wf_v1_1.cwl"))
+            .resolve()
+            .as_uri()
+        )
+        cwl_obj = load_document_by_uri(uri)
+        cwl_utils.parser.utils.static_checker(cwl_obj)
+
+    with pytest.raises(ValidationException):
+        uri = (
+            Path(get_data("testdata/count-lines6-single-source-wf_v1_2.cwl"))
+            .resolve()
+            .as_uri()
+        )
+        cwl_obj = load_document_by_uri(uri)
+        cwl_utils.parser.utils.static_checker(cwl_obj)
+
+    with pytest.raises(ValidationException):
+        uri = (
+            Path(get_data("testdata/cond-single-source-wf-003.1.cwl"))
+            .resolve()
+            .as_uri()
+        )
+        cwl_obj = load_document_by_uri(uri)
+        cwl_utils.parser.utils.static_checker(cwl_obj)
+
+    with pytest.raises(ValidationException):
+        uri = (
+            Path(get_data("testdata/cond-single-source-wf-004.1.cwl"))
+            .resolve()
+            .as_uri()
+        )
+        cwl_obj = load_document_by_uri(uri)
+        cwl_utils.parser.utils.static_checker(cwl_obj)
+
+
+def test_static_checker_success() -> None:
+    """Confirm that static type checker correctly validates workflows."""
```

Comment:
Can you parameterize these so the test is not so repetitive?

https://docs.pytest.org/en/6.2.x/parametrize.html

Reply:
I did it